File: win-prog/ProgWinVB/Dialog Boxes/AboutBox/AboutBox.vb
'-----------------------------------------
' AboutBox.vb (c) 2002 by Charles Petzold
'-----------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class AboutBox
    Inherits Form

    Shared Sub Main()
        Application.Run(New AboutBox())
    End Sub

    Sub New()
        Text = "About Box"
        Icon = New Icon(Me.GetType(), "AforAbout.ico")

        Menu = New MainMenu()
        Menu.MenuItems.Add("&Help")
        Menu.MenuItems(0).MenuItems.Add("&About AboutBox...", _
                                  AddressOf MenuAboutOnClick)
    End Sub

    Sub MenuAboutOnClick(ByVal obj As Object, ByVal ea As EventArgs)
        Dim dlg As New AboutDialogBox()
        dlg.ShowDialog()
    End Sub
End Class

Class AboutDialogBox
    Inherits Form

    Sub New()
        Text = "About AboutBox"

        StartPosition = FormStartPosition.CenterParent
        FormBorderStyle = FormBorderStyle.FixedDialog
        ControlBox = False
        MaximizeBox = False
        MinimizeBox = False
        ShowInTaskbar = False

        ' Create first label with program name.
        Dim lbl1 As New Label()
        lbl1.Parent = Me
        lbl1.Text = " AboutBox Version 1.0 "
        lbl1.Font = New Font(FontFamily.GenericSerif, 24, _
                             FontStyle.Italic)
        lbl1.AutoSize = True
        lbl1.TextAlign = ContentAlignment.MiddleCenter

        ' Create picture box containing icon.
        Dim icon As New Icon(Me.GetType(), "AforAbout.ico")
        Dim picbox As New PictureBox()
        picbox.Parent = Me
        picbox.Image = icon.ToBitmap()
        picbox.SizeMode = PictureBoxSizeMode.AutoSize
        picbox.Location = New Point(lbl1.Font.Height \ 2, _
                                    lbl1.Font.Height \ 2)

        lbl1.Location = New Point(picbox.Right, lbl1.Font.Height \ 2)

        Dim iClientWidth As Integer = lbl1.Right

        ' Create second label with copyright and link.
        Dim lnklbl2 As New LinkLabel()
        lnklbl2.Parent = Me
        lnklbl2.Text = Chr(169) & " 2002 by Charles Petzold"
        lnklbl2.Font = New Font(FontFamily.GenericSerif, 16)
        lnklbl2.Location = New Point(0, lbl1.Bottom + lnklbl2.Font.Height)
        lnklbl2.Size = New Size(iClientWidth, lnklbl2.Font.Height)
        lnklbl2.TextAlign = ContentAlignment.MiddleCenter

        ' Set link area and event handler.
        lnklbl2.LinkArea = New LinkArea(10, 15)
        AddHandler lnklbl2.LinkClicked, AddressOf LinkLabelOnLinkClicked

        ' Create OK button.
        Dim btn As New Button()
        btn.Parent = Me
        btn.Text = "OK"
        btn.Size = New Size(4 * btn.Font.Height, _
                            2 * btn.Font.Height)
        btn.Location = New Point((iClientWidth - btn.Size.Width) \ 2, _
                                 lnklbl2.Bottom + 2 * btn.Font.Height)
        btn.DialogResult = DialogResult.OK
        btn.TabIndex = 0

        CancelButton = btn
        AcceptButton = btn
        ClientSize = New Size(iClientWidth, _
                              btn.Bottom + 2 * btn.Font.Height)
    End Sub

    Private Sub LinkLabelOnLinkClicked(ByVal obj As Object, _
            ByVal lllcea As LinkLabelLinkClickedEventArgs)
        System.Diagnostics.Process.Start("http://www.charlespetzold.com")
    End Sub
End Class

File: win-prog/ProgWinCS/Dialog Boxes/AboutBox/AboutBox.cs
//---------------------------------------
// AboutBox.cs © 2001 by Charles Petzold
//---------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class AboutBox: Form
{
     public static void Main()
     {
          Application.Run(new AboutBox());
     }
     public AboutBox()
     {
          Text = "About Box";
          Icon = new Icon(GetType(), "AboutBox.AforAbout.ico");

          Menu = new MainMenu();
          Menu.MenuItems.Add("&Help");
          Menu.MenuItems[0].MenuItems.Add("&About AboutBox...",
                                        new EventHandler(MenuAboutOnClick));
     }
     void MenuAboutOnClick(object obj, EventArgs ea)
     {
          AboutDialogBox dlg = new AboutDialogBox();
          dlg.ShowDialog();
     }
}
class AboutDialogBox: Form
{
     public AboutDialogBox()
     {
          Text = "About AboutBox";

          StartPosition   = FormStartPosition.CenterParent;
          FormBorderStyle = FormBorderStyle.FixedDialog;
          ControlBox      = false;
          MaximizeBox     = false;
          MinimizeBox     = false;
          ShowInTaskbar   = false;

          Label label1     = new Label();
          label1.Parent    = this;
          label1.Text      = " AboutBox Version 1.0 ";
          label1.Font      = new Font(FontFamily.GenericSerif, 24, 
                                      FontStyle.Italic);
          label1.AutoSize  = true;
          label1.TextAlign = ContentAlignment.MiddleCenter;

          Icon icon = new Icon(GetType(), "AboutBox.AforAbout.ico");

          PictureBox picbox = new PictureBox();
          picbox.Parent     = this;
          picbox.Image      = icon.ToBitmap();
          picbox.SizeMode   = PictureBoxSizeMode.AutoSize;
          picbox.Location   = new Point(label1.Font.Height / 2, 
                                        label1.Font.Height / 2);

          label1.Location  = new Point(picbox.Right,label1.Font.Height / 2);

          int iClientWidth = label1.Right;

          Label label2     = new Label();
          label2.Parent    = this;
          label2.Text      = "\x00A9 2001 by Charles Petzold";
          label2.Font      = new Font(FontFamily.GenericSerif, 16);
          label2.Location  = new Point(0, label1.Bottom + 
                                          label2.Font.Height);
          label2.Size      = new Size(iClientWidth, label2.Font.Height);
          label2.TextAlign = ContentAlignment.MiddleCenter;

          Button button   = new Button();
          button.Parent   = this;
          button.Text     = "OK";
          button.Size     = new Size(4 * button.Font.Height, 
                                     2 * button.Font.Height);
          button.Location = new Point((iClientWidth - button.Size.Width) / 2,
                                   label2.Bottom + 2 * button.Font.Height);

          button.DialogResult = DialogResult.OK;

          CancelButton = button;
          AcceptButton = button;

          ClientSize = new Size(iClientWidth, 
                                button.Bottom + 2 * button.Font.Height);
     }
}